src/iosrv/srv.c

```diff
@@ -1132,8 +1129,27 @@ dss_srv_init()
 	}
 	xstream_data.xd_init_step = XD_INIT_ULT_BARRIER;
 
-	rc = bio_nvme_init(dss_storage_path, dss_nvme_conf, dss_nvme_shm_id,
-		dss_nvme_mem_size);
+	/* register xstream-local storage key */
+	rc = pthread_key_create(&dss_tls_key, NULL);
+	if (rc) {
+		rc = dss_abterr2der(rc);
+		D_GOTO(failed, rc);
+	}
+	xstream_data.xd_init_step = XD_INIT_TLS_REG;
+
+	/* initialize xstream-local storage */
+	xstream_data.xd_dtc = dss_tls_init(DAOS_SERVER_TAG);
```

Comment:
Why TLS init is moved here? I guess it because vos_db_init() relies on TLS? Can we workaround the problem in some other way?

The problem here is that xstream isn't even created yet, I think that TLS allocation needs be done in the xstream (like current dss_srv_handler()) to ensure the memory allocation close to the binding core.

Reply:
It is not moved to here, I just create TLS for main() pthread, it is not performance sensitive and dss_srv_handler() still allocate TLS for I/O xstream.

Reply:
Ok, I see. Thanks.